### PR TITLE
octavia: Add topology setting (SOC-10876)

### PR DESCRIPTION
**Pull request by nicolasbock**
Currently a user can not deploy Octavia in active/standby topology
because Crowbar does not allow changes to the corresponding setting in
the Octavia configuration file. This change makes the setting
configurable.

Signed-off-by: Nicolas Bock <nicolas.bock@suse.com>

### CODE REVIEW

File: chef/data_bags/crowbar/migrate/octavia/302_add_loadbalancer_topology.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+def upgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs[:amphora][:loadbalancer_topology] = template_deployment[:amphora][:loadbalancer_topology] \
+    unless attrs.key? :loadbalancer_topology
```
Comment (stefannica) [inline chef/data_bags/crowbar/migrate/octavia/302_add_loadbalancer_topology.rb]: I think you mean `unless attrs.key? :amphora && attrs[:amphora].key? :loadbalancer_topology`

Comment (nicolasbock) [reply chef/data_bags/crowbar/migrate/octavia/302_add_loadbalancer_topology.rb]: Ah right :smile: But can't I assume that the `amphora` key is there?

File: chef/data_bags/crowbar/migrate/octavia/302_add_loadbalancer_topology.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  attrs[:amphora][:spare_amphora_pool_size] = template_deployment[:amphora][:spare_amphora_pool_size] \
```
[the remark this answers is not in the document]
Comment (nicolasbock) [reply chef/data_bags/crowbar/migrate/octavia/302_add_loadbalancer_topology.rb]: Not sure how to best shorten that line. Do we care?

**Comment by stefannica** [comment]
needs a rebase

**Comment by nicolasbock** [comment]
Rebased and added a validation step for the topology.

**Comment by yarunachalam** [comment]
https://ci.nue.suse.com/job/openstack-crowbar/683/ - validated using this repo
http://download.suse.de/ibs/home:/nicolasbock:/branches:/Devel:/Cloud:/9:/Staging/SLE_12_SP4/
it works as expected for ACTIVE_STANDBY topology setting for new version keepalived-2.0.19-2.1.x86_64

**Pull request merged**